Packs/SecurityIntelligenceServicesFeed/Integrations/SecurityIntelligenceServicesFeed/SecurityIntelligenceServicesFeed.py

```diff
@@ -0,0 +1,540 @@
+
+        except botocore.exceptions.HTTPClientError as exception:
+            raise ValueError(MESSAGES['HTTP_CLIENT_ERROR'] + str(exception))
+
+        except Exception as exception:
+            raise ValueError(MESSAGES['ERROR'] + str(exception))
+
+
+''' HELPER FUNCTIONS '''
+
+
+def validate_feeds(feed_types: List[str]) -> None:
+    """
+    Checks that given feeds are exist or not.
+    If feed_types is empty then throws a ValueError.
+    If any feed_type is not exist in BUCKET throws a ValueError.
+
+    :param feed_types: Feed types provided.
+    :return: True if feeds exist else return error.
+    """
+    if not feed_types:
+        raise ValueError(MESSAGES['REQUIRED_FEED_TYPE_ERROR'])
+    for feed in feed_types:
+        if feed not in BUCKETS:
+            raise ValueError(MESSAGES['INVALID_FEED_TYPE_ERROR'])
+
+
+def get_last_key_from_integration_context(feed_type: str) -> str:
+    """
+    To get last fetched key of feed from integration context.
+
+    :param feed_type: Type of feed to get last fetched key.
+    :return: list of S3 object keys.
+    """
+    feed_context = demisto.getIntegrationContext().get('SISContext', [])
+    for cached_feed in feed_context:
+        cached_key = cached_feed.get(feed_type, '')
+        if cached_key:
+            return cached_key
+    return ''
+
+
+def set_last_key_to_integration_context(feed_type: str, key: str) -> None:
+    """
+    To set last fetched key of feed to integration context.
+
+    :param feed_type: Type of feed to set.
+    :param key: Key to set.
+    :return: None
+    """
+    feed_context = demisto.getIntegrationContext().get('SISContext', [])
+    for f_type_dict in feed_context:
+        if f_type_dict.get(feed_type, ''):
+            f_type_dict[feed_type] = key
+            demisto.setIntegrationContext({'SISContext': feed_context})
+            return
+    feed_context.append({feed_type: key})
+    demisto.setIntegrationContext({'SISContext': feed_context})
+
+
+def validate_limit(limit: str) -> None:
+    """
+    Validates the limit argument.
+    If the limit is 0 >= limit > 1000 or not an integer then throws a ValueError.
+
+    :param limit: Number of feeds to fetch .
+    :return: None
+    """
+    try:
+        val = int(limit)
+        if val <= 0 or val > 1000:
+            raise ValueError
+    except ValueError:
+        raise ValueError(MESSAGES['INVALID_LIMIT_ERROR'])
+
+
+def indicator_field_mapping(feed_type: str, indicator: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Maps the indicator fields.
+
+    :param feed_type: Type of feed.
+    :param indicator: Indicator dictionary.
+    :return: Dict of fields.
+    """
+    fields: Dict[str, Any] = {
+        'service': 'Passive Total',
+    }
+    try:
+        if feed_type in ['host', 'domain']:
+            if indicator.get('Timestamp'):
+                fields['firstseenbysource'] = datetime.fromtimestamp(int(indicator['Timestamp'])).isoformat() + 'Z'
```

Comment:
Can you show an indicator data example? how do you know that this date is in UTC?

Reply:
These timestamps are in epoch format in data. To be sure that parsed time is in UTC, we are changing to "datetime.utcfromtimestamp"

Reply:
I'd rather make this an aware datetime object and this why you can remove the 'Z' and completely trust the `isoformat` method.

Try: 
```
datetime.fromtimestamp(int(indicator['Timestamp']), ,datetime.timezone.utc).isoformat()
```
and don't forget to import the `timezone` from datetime (i.e `from datetime import datetime, timezone`)

Reply:
Same goes to all date parsing

Reply:
Thanks @hod-alpert, we will make changes to all date-time parsing to have aware Datetime object as you suggested.